plugins/drivers/driver.go

```diff
@@ -90,7 +90,7 @@ type ExecOptions struct {
 // network namespace for which tasks can join. This only needs to be implemented
 // if the driver MUST create the network namespace
 type DriverNetworkManager interface {
-	CreateNetwork(allocID string) (*NetworkIsolationSpec, bool, error)
+	CreateNetwork(request *NetworkCreateRequest) (*NetworkIsolationSpec, bool, error)
```

Comment:
So I like the new signature better than the old, but I'm concerned about backward compatibility. I was hoping there would be a way for our grpc layer to keep the old field and fill in the allocID from the struct... but I don't see it now.

The other route to backward compatibility would be to add the new struct with only the Hostname field, and leave the allocID in place. Not nearly as tidy as the new parameter sadly.